cookbook/security/multiple_guard_authenticators.rst

```diff
@@ -0,0 +1,168 @@
+
+    .. code-block:: xml
+
+        <!-- app/config/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <config>
+                <!-- ... -->
+                <firewall name="default">
+                    <anonymous />
+                    <guard entry_point="app.form_login_authenticator">
+                        <authenticator>app.form_login_authenticator</authenticator>
+                        <authenticator>app.facebook_connect_authenticator</authenticator>
+                    </guard>
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // app/config/security.php
+        $container->loadFromExtension('security', array(
+            // ...
+            'firewalls' => array(
+                'default' => array(
+                    'anonymous' => null,
+                    'guard' => array(
+                        'entry_point' => 'app.form_login_authenticator',
+                        'authenticators' => array(
+                            'app.form_login_authenticator',
+                            'app.facebook_connect_authenticator'
+                        ),
+                    ),
+                ),
+            ),
+        ));
+
+There is one limitation with this approach - you have to use exactly one entry point.
+
+Multiple authenticators with separate entry points
+--------------------------------------------------
+
+Let's now have an example of two different authenticators: one based on login form, another one on an API token.
+When user hits secured area he should be redirected to the login page.
+Also when user hits an API endpoint, he should get a relevant API response.
+
+Solution for this use case is to provide guard authenticators in two separate firewalls.
+
+This is an example of your configuration:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/security.yml
+        security:
+            # ...
+            firewalls:
+                api:
+                    pattern: ^/api/
+                    guard:
+                        authenticators:
+                            - app.api_token_authenticator
+                default:
+                    anonymous: ~
+                    guard:
+                        authenticators:
+                            - app.form_login_authenticator
+            access_control:
+                - { path: ^/login, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+                - { path: ^/api, roles: ROLE_API_USER }
+                - { path: ^/, roles: ROLE_ADMIN }
```

Comment:
this means the whole site can only be accessed by its admins. What about changing it to `ROLE_USER` or `^/admin`?